#### Задача 1. 
Когда используется критерий Стьюдента, а когда Z –критерий?

Критерий Стьюдента (t - критерий)используется для сравнения средних значений двух групп, когда стандартное отклонение совокупности неизвестно, а Z-критерий используется, когда известно стандартное отклонение совокупности и размер выборки большой.
T-тесты основаны на t-распределении, которое используется для меньших размеров выборки и неизвестных стандартных отклонений населения, тогда как Z-тесты используют стандартное нормальное распределение.

#### Задача 2. 
Проведите тест гипотезы. Утверждается, что шарики для подшипников, изготовленные
автоматическим станком, имеют средний диаметр 17 мм.
Используя односторонний критерий с α=0,05, проверить эту гипотезу, если в выборке из
n=100 шариков средний диаметр
оказался равным 17.5 мм, а дисперсия известна и равна 4 кв. мм.

Н0 - шарики для подшипников, изготовленные
автоматическим станком, имеют средний диаметр 17 мм
Н1 - средний диаметр шариков не равен 17 мм


In [70]:
import numpy as np
from scipy import stats

In [71]:
# Известна дисперсия, значит и сигма, значит используем z - критерий
z = (17.5 - 17) / (2/100**0.5)
z

2.5

In [72]:
alpha = 0.05
stats.norm.ppf(1 - alpha)

1.6448536269514722

Наблюдаемое значение больше табличного. Значит отвергаем гипотезу H0. Принимаем гипотезу Н1

#### Задача 3. 
Проведите тест гипотезы. Продавец утверждает, что средний вес пачки печенья
составляет 200 г.
Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет:
202, 203, 199, 197, 195, 201, 200, 204, 194, 190.
Известно, что их веса распределены нормально.
Верно ли утверждение продавца, если учитывать, что доверительная вероятность равна
99%? (Провести двусторонний тест.)

In [73]:
x = np.array([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])
x_mean = x.mean()
x_std = x.std(ddof = 1)

t = (x_mean - 200) / (x_std / 10**0.5)
x_mean, x_std, t

(198.5, 4.453463071962462, -1.0651074037450896)

H0 - средний вес пачки печенья составляет 200 г.
Н1 - средний вес пачки печенья не равен 200 г.

In [74]:

alfa = 0.01
t1 = stats.t.ppf(alfa / 2, df = 10 - 1)
t2 = stats.t.ppf(1 - alfa / 2, df = 10 - 1)
t1, t2

(-3.2498355440153697, 3.2498355440153697)

t попадает внутрь интервала между t1 и t2. Значит принимаем гипотезу Н0. Средний вес пачки печенья 200 г.

#### Задача 4. 
Есть ли статистически значимые различия в росте дочерей?
Рост матерей 172, 177, 158, 170, 178,175, 164, 160, 169, 165
Рост взрослых дочерей: 173, 175, 162, 174, 175, 168, 155, 170, 160, 163

In [75]:
m = np.array([172, 177, 158, 170, 178,175, 164, 160, 169, 165])
d = np.array([173, 175, 162, 174, 175, 168, 155, 170, 160, 163])
stats.ttest_rel(m, d)

TtestResult(statistic=0.6648478531431979, pvalue=0.5228168632983574, df=9)

pvalue(0.5228168632983574) > alfa (0.05), значит Н0 не отвергается, статистически значимых различий не обнаружено на уровне значимости alpha = 0.05